src/main/scala/de/upb/cs/swt/delphi/crawler/instancemanagement/InstanceRegistry.scala

```diff
@@ -0,0 +1,166 @@
+      if(response.status == StatusCodes.OK){
+        Await.result(Unmarshal(response.entity).to[String] map { assignedID =>
+          val id = assignedID.toLong
+          log.info(s"Successfully registered at Instance Registry, got ID $id.")
+          Success(id)
+        } recover { case ex =>
+          log.warning(s"Failed to read assigned ID from Instance Registry, exception: $ex")
+          Failure(ex)
+        }, Duration.Inf)
+      }
+      else {
+        val statuscode = response.status
+        log.warning(s"Failed to register at Instance Registry, server returned $statuscode")
+        Failure(new RuntimeException(s"Failed to register at Instance Registry, server returned $statuscode"))
+      }
+
+    } recover {case ex =>
+      log.warning(s"Failed to register at Instance Registry, exception: $ex")
+      Failure(ex)
+    }, Duration.Inf)
+  }
+
+  def retrieveElasticSearchInstance(configuration: Configuration) : Try[Instance] = {
+    if(!configuration.usingInstanceRegistry) {
+      Failure(new RuntimeException("Cannot get ElasticSearch instance from Instance Registry, no Instance Registry available."))
+    } else {
+      val request = HttpRequest(method = HttpMethods.GET, configuration.instanceRegistryUri + "/matchingInstance?ComponentType=ElasticSearch")
+
+      Await.result(Http(system).singleRequest(request) map {response =>
+        val status = response.status
+        if(status == StatusCodes.OK) {
+
+          Await.result(Unmarshal(response.entity).to[Instance] map {instance =>
+            val elasticIP = instance.iP
+            log.info(s"Instance Registry assigned ElasticSearch instance at ${elasticIP.getOrElse("None")}")
+            Success(instance)
+          } recover {case ex =>
+            log.warning(s"Failed to read response from Instance Registry, exception: $ex")
+            Failure(ex)
+          }, Duration.Inf)
+        }
+        else{
+          log.warning(s"Failed to read response from Instance Registry, server returned $status")
+          Failure(new RuntimeException(s"Failed to read response from Instance Registry, server returned $status"))
+        }
+      } recover { case ex =>
+        log.warning(s"Failed to request ElasticSearch instance from Instance Registry, exception: $ex ")
+        Failure(ex)
+      }, Duration.Inf)
+    }
+  }
+
+  def sendMatchingResult(isElasticSearchReachable : Boolean, configuration: Configuration) : Try[Unit] = {
+    if(!configuration.usingInstanceRegistry) {
+      Failure(new RuntimeException("Cannot post matching result to Instance Registry, no Instance Registry available."))
+    } else {
+      if(configuration.elasticsearchInstance.iD.isEmpty) {
+        Failure(new RuntimeException("Cannot post matching result to Instance Registry, assigned ElasticSearch instance has no ID."))
+      } else {
+        val idToPost = configuration.elasticsearchInstance.iD.getOrElse(-1L)
```

Comment:
Is the -1 case handled inside of the instance registry?

Reply:
This `.getOrElse` was introduced to avoid the codestyle warnings that resulted from using `.get` on Options. In fact the check right above this code will make sure that the Option is not empty, therefore the -1 case is only meant to provoke an BadRequest response from the Instance Registry if ever executed.